clippy_lints/src/redundant_async_block.rs

```diff
@@ -48,6 +48,11 @@ impl EarlyLintPass for RedundantAsyncBlock {
             !future.span.from_expansion() &&
             !await_in_expr(future)
         {
+            if captured_by_ref(last) {
```

Comment:
I guess that this should also check the async capture mode:

```suggestion
            if CaptureBy::Ref = cap_kind && captured_by_ref(last) {
```

The `cap_kind` can be taken from `ExprKind::Async(cap_kind, _, block)` in line 45

---

When looking at the tests, it looks like this also detects cases with `async move`. Is this intentional? In that case, my suggestion would be wrong, but the function name would also be misleading. `captured_by_move` would then be better or if the `CaptureBy` doesn't matter at all, just something like `captures_value` :)

Reply:
Woops, this suggestion has already been made above. Turns out I should have read everything before. In that case I would suggest calling the function `captures_value`

Reply:
Renamed.